src/libs/search.ts

```diff
@@ -0,0 +1,217 @@
+import type {
+	ScoreAndHighlightResultItem,
+	ScoreResultItem,
+	SearchIndexes,
+	SearchResults,
+	HighlightedText
+} from '$models/search';
+
+/**
+ * Tokenizes the given text into an array of words.
+ * If the Intl.Segmenter API is available, it will be used to segment the text into words.
+ * Otherwise, the text will be split by spaces.
+ * @param text - The text to tokenize.
+ * @returns An array of words.
+ */
+const tokenize = (text: string) => {
+	// Use Intl.Segmenter API if available
+	if (Intl !== undefined && Intl.Segmenter !== undefined) {
```

Comment:
`undefined` is already falsy (treated as false in a condition) and I think you can use optional chaining to be more concise
```js
if (Intl?.Segmenter) {
```

Reply:
I think it's more readable, but `Intl && Intl.Segmenter` is also a good alternative.